mllam_data_prep/derived_variables.py

```diff
@@ -0,0 +1,478 @@
+        if isinstance(derived_field, xr.DataArray):
+            derived_field = _check_attributes(
+                derived_field, derived_variable_attributes
+            )
+            ds_derived_vars[derived_field.name] = derived_field
+        elif isinstance(derived_field, tuple) and all(
+            isinstance(field, xr.DataArray) for field in derived_field
+        ):
+            for field in derived_field:
+                field = _check_attributes(field, derived_variable_attributes)
+                ds_derived_vars[field.name] = field
+        else:
+            raise TypeError(
+                "Expected an instance of xr.DataArray or tuple(xr.DataArray),"
+                f" but got {type(derived_field)}."
+            )
+
+        # Add back dropped coordinates
+        ds_derived_vars = _return_dropped_coordinates(
+            ds_derived_vars, ds_input, required_coordinates, chunks
+        )
+
+    return ds_derived_vars
+
+
+def _chunk_dataset(ds, chunks):
+    """
+    Chunk dataset and check the chunk size.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+        Dataset to be chunked
+    chunks: Dict[str, int]
+        Dictionary with keys as dimensions to be chunked and
+        chunk sizes as the values
+
+    Returns
+    -------
+    ds: xr.Dataset
+        Dataset with chunking applied
+    """
+    # Define the memory limit check
+    memory_limit_check = 1 * 1024**3  # 1 GB
+
+    # Check the chunk size
+    for var_name, var_data in ds.data_vars.items():
+        total_size = 1
+
+        for dim, chunk_size in chunks.items():
+            dim_size = ds.sizes.get(dim, None)
+            if dim_size is None:
+                raise KeyError(f"Dimension '{dim}' not found in the dataset.")
+            total_size *= chunk_size
+
+        dtype = var_data.dtype
+        bytes_per_element = np.dtype(dtype).itemsize
+
+        memory_usage = total_size * bytes_per_element
+
+        if memory_usage > memory_limit_check:
+            logger.warning(
+                f"The chunk size for '{var_name}' exceeds '{memory_limit_check}' GB."
+            )
+
+    # Try chunking
+    try:
+        ds = ds.chunk(chunks)
+    except Exception as ex:
+        raise Exception(f"Error chunking dataset: {ex}")
+
+    return ds
+
+
+def _get_derived_variable_function(function_namespace):
+    """
+    Function for getting the function for deriving
+    the specified variable.
+
+    Parameters
+    ----------
+    function_namespace: str
+        The full function namespace or just the function name
+        if it is a function included in this module.
+
+    Returns
+    -------
+    function: object
+        Function for deriving the specified variable
+    """
+    # Get the name of the calling module
+    calling_module = globals()["__name__"]
+
+    # Get module and function names
+    function_namespace_list = function_namespace.rsplit(".")
+    if len(function_namespace_list) > 1:
```

Comment:
You can actually avoid this whole if-statement if you do
`module_name, _, function_name = function_namespace.rpartition(".")

Reply:
Rather than using `globals()` which would require that the module is already imported I think it might be better to use the `importlib` package.

I think you would write:

```python
module = importlib.import_module(module_name)
fn = getattr(module, function_name)
```